dbms/src/Functions/tests/gtest_strings_reverse.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsString.h>
+#include <Interpreters/Context.h>
+#include <TestUtils/FunctionTestUtils.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+#include <string>
+#include <vector>
+
+#pragma GCC diagnostic pop
+
+namespace DB::tests
+{
+class StringReverse : public DB::tests::FunctionTest
+{
+protected:
+    static ColumnWithTypeAndName toVec(const std::vector<String> & v)
+    {
+        return createColumn<String>(v);
+    }
+
+    static ColumnWithTypeAndName toNullableVec(const std::vector<std::optional<String>> & v)
+    {
+        return createColumn<Nullable<String>>(v);
+    }
+
+    static ColumnWithTypeAndName toConst(const String & s)
+    {
+        return createConstColumn<String>(1, s);
+    }
+};
+// test reverse
+TEST_F(StringReverse, stringReverseTest)
```

Comment:
Maybe you can refactor the TEST function like:
```cpp
// test reverse
TEST_F(StringReverse, stringReverseTest)
try
{
    // test vector
    code...
    // test nullable
    code...
    // test const
    code...
    // test null
    code...
}
CATCH

// test reverseUTF8
TEST_F(StringReverse, stringReverseUTF8Test)
try
{
    // test vector
    code...
    // test nullable
    code...
    // test const
    code...
    // test null
    code...
}
CATCH
```
This makes the test code structure more clearer